compiler/p/codegen/OMRRealRegister.cpp

```diff
@@ -186,7 +186,14 @@ OMR::Power::RealRegister::setRegisterFieldRA(uint32_t *instruction)
    else
       *instruction |= fullRegBinaryEncodings[_registerNumber] << pos_RA;
    }
-
+void
+OMR::Power::RealRegister::setRegisterFieldRB(uint32_t *instruction)
+  {
+      if (self()->isConditionRegister())
+        *instruction |= fullRegBinaryEncodings[_registerNumber] << (pos_RB + 2);
+      else
+         *instruction |= fullRegBinaryEncodings[_registerNumber] << pos_RB;
+  }
```

Comment:
Am I wrong in assuming something like the following will generate smaller code on Power?  One wrinkle is the shift amount is no longer constant...
```
   int32_t RBshift = self()->isConditionRegister() ? pos_RB + 2 : pos_RB;
   *instruction |= fullRegBinaryEncodings[_registerNumber] << RBshift;
```

Reply:
Some compilers may common fullRegBinaryEncodings[_registerNumber] between the two branches for the sake of space (not run time).

Reply:
Thanks for clarifying.